### PR TITLE
Add libSyntax support for #chainableGradient.

### PR DESCRIPTION
**Pull request by rxwei**
As a follow-up on #18250, this patch adds libSyntax support and fixes the naming in syntax nodes for reverse-mode differential operators.

Also rename `GradientExpr` to `ReverseAutoDiffExpr` in syntax nodes, since it is supposed to generalize the C++ class `swift::ReverseAutoDiffExpr` instead of its subclass `swift::GradientExpr`.

### CODE REVIEW

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Review by dan-zheng — APPROVED** [review]

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

File: utils/gyb_syntax_support/ExprNodes.py [hunk 1]
```diff
@@ -571,14 +571,14 @@
...
-    # e.g. #gradient(foo(_:_:), wrt: .0, .1)
-    Node('GradientExpr', kind='Expr',
+    # e.g. "#gradient(foo(_:_:), wrt: .0, .1)"
+    Node('ReverseAutoDiffExpr', kind='Expr',
          traits=['Parenthesized'],
          children=[
              Child('Identifier', kind='Token',
                    token_choices=[
                        'PoundGradientToken',
+                       'ChainableGradientToken',
```
Comment (dan-zheng) [inline utils/gyb_syntax_support/ExprNodes.py]: Do you mean `PoundChainableGradientToken`?

Comment (rxwei) [reply utils/gyb_syntax_support/ExprNodes.py]: Aha. Good catch.

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Pull request merged**